lib/RestService/src/RestService.cpp

```diff
@@ -0,0 +1,394 @@
+        handleAsyncWebResponse(restId, rsp);
+    };
+    AsyncHttpClient::OnError errCallback = [this](void* restId) {
+        handleFailedWebRequest(restId);
+    };
+    AsyncHttpClient::OnClosed closedCallback = [this]() {
+        m_isWaitingForResponse = false;
+    };
+
+    isSuccessful = m_cmdQueue.create(CMD_QUEUE_SIZE);
+
+    if (true == isSuccessful)
+    {
+        m_client.regOnResponse(rspCallback);
+        m_client.regOnError(errCallback);
+        m_client.regOnClosed(closedCallback);
+    }
+
+    return isSuccessful;
+}
+
+void RestService::stop()
+{
+    m_client.regOnResponse(nullptr);
+    m_client.regOnError(nullptr);
+    m_client.regOnClosed(nullptr);
+
+    m_cmdQueue.destroy();
+}
+
+void RestService::process()
+{
+    if (false == m_isWaitingForResponse)
+    {
+        Cmd cmd;
+
+        if (true == m_cmdQueue.receive(&cmd, 0U))
+        {
+            m_isWaitingForResponse = true;
+
+            if (true == m_client.begin(String(cmd.url)))
+            {
+                switch (cmd.id)
+                {
+                case CMD_ID_GET:
+                    if (false == m_client.GET(cmd.restId))
+                    {
+                        Msg msg;
+
+                        msg.restId = cmd.restId;
+                        msg.isMsg  = false;
+                        msg.rsp    = nullptr;
+
+                        if (false == this->m_taskProxy.send(msg))
+                        {
+                            LOG_ERROR("Msg could not be sent to Msg-Queue");
+                        }
+
+                        m_isWaitingForResponse = false;
+                    }
+                    break;
+
+                case CMD_ID_POST:
+                    if (false == m_client.POST(cmd.u.data.data, cmd.u.data.size, cmd.restId))
+                    {
+                        Msg msg;
+
+                        msg.restId = cmd.restId;
+                        msg.isMsg  = false;
+                        msg.rsp    = nullptr;
+
+                        if (false == this->m_taskProxy.send(msg))
+                        {
+                            LOG_ERROR("Msg could not be sent to Msg-Queue");
+                        }
+
+                        m_isWaitingForResponse = false;
+                    }
+                    break;
+
+                default:
+                    break;
+                };
+            }
+            else
+            {
+                Msg msg;
+
+                msg.restId = cmd.restId;
+                msg.isMsg  = false;
+                msg.rsp    = nullptr;
+
+                if (false == this->m_taskProxy.send(msg))
+                {
+                    LOG_ERROR("URL could not be parsed");
+                }
+
+                m_isWaitingForResponse = false;
+            }
+        }
+    }
+}
+
+void RestService::setCallback(void* restId, PreProcessCallback preProcessCallback)
```

Comment:
RestID is still strange, by using the address of a variable, which might change in the application. Shall be solved later in a separate PR.